src/GitHub.App/ViewModels/GistCreationViewModel.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using GitHub.Exports;
+using System.ComponentModel.Composition;
+using System.Reactive.Linq;
+using GitHub.Api;
+using GitHub.Factories;
+using GitHub.Primitives;
+using GitHub.Services;
+using Octokit;
+using ReactiveUI;
+using NullGuard;
+
+namespace GitHub.ViewModels
+{
+    [ExportViewModel(ViewType=UIViewType.Gist)]
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    public class GistCreationViewModel : BaseViewModel, IGistCreationViewModel
+    {
+        readonly ISelectedTextProvider selectedTextProvider;
+        readonly IApiClient apiClient;
+
+        [ImportingConstructor]
+        public GistCreationViewModel(ISelectedTextProvider selectedTextProvider, IApiClientFactory apiClientFactory)
+        {
+            Title = Resources.CreateGistTitle;
+            this.selectedTextProvider = selectedTextProvider;
+            this.apiClient = apiClientFactory.Create(HostAddress.GitHubDotComHostAddress);
```

Comment:
Of course, if there's no connection that matches the `ActiveRepo` (or if `ActiveRepo` is null), passing null to `RunUI` is fine, it'll just have the user login to whatever they want, and then save that `IConnection` so that the view model can grab it when it loads from the (rather convoluted) constructor.